gnd/src/main/java/com/google/android/gnd/ui/map/Extent.java

```diff
@@ -0,0 +1,69 @@
+package com.google.android.gnd.ui.map;
+
+import androidx.annotation.Nullable;
+
+import com.google.android.gnd.model.basemap.tile.Tile;
+import com.google.auto.value.AutoValue;
+
+@AutoValue
+public abstract class Extent {
+  public enum State {
+    DOWNLOADED,
+    PENDING_DOWNLOAD,
+    PENDING_REMOVAL,
+    NONE
+  }
+
+  public abstract String getId();
+
+  public abstract State getState();
+
+  public static Builder newBuilder() {
+    return new AutoValue_Extent.Builder();
+  }
+
+  public abstract Builder toBuilder();
+
+  public static Extent fromTile(Tile tile) {
+    switch (tile.getState()) {
+      case IN_PROGRESS:
+        return Extent.newBuilder().setId(tile.getId()).setState(State.PENDING_DOWNLOAD).build();
```

Comment:
This can be made more DRY be refactoring our the mapping from tile states to State (e.g., fromTileState()).

Reply:
Thanks for the recommendation, done!